vaadin-map-flow-parent/vaadin-map-flow/src/main/resources/META-INF/resources/frontend/vaadin-map/mapConnector.js

```diff
@@ -0,0 +1,66 @@
+import { synchronize } from "./synchronization";
+
+(function () {
+  function init(mapElement) {
+    mapElement.$connector = {
+      /**
+       * Synchronize a configuration object into the internal OpenLayers map instance.
+       *
+       * The target parameter can be left unspecified, in which case the OpenLayers
+       * map instance itself is used as synchronization target. This results in a
+       * full sync of the map configuration hierarchy.
+       * Specifying a target instance allows synchronizing only a part of the
+       * configuration. In that case the configuration parameter needs to be for the
+       * specific part that should be synchronized. For example, when passing an
+       * `ol/View` instance as target, then the configuration parameter should
+       * contain a configuration object for a view as well.
+       *
+       * @param configuration the configuration object to synchronize from
+       * @param target the OpenLayers configuration instance to synchronize into, or undefined if the root map instance should be used
+       */
+      synchronize(configuration, target) {
+        if (!configuration || typeof configuration !== "object") {
+          throw new Error("Configuration must be an object");
+        }
+        // Assume we want to sync the full map by default
+        if (!target) {
+          target = mapElement.configuration;
+        }
+
+        // We don't want the synchronization to return a new instance (we have no
+        // idea where to store it afterwards), so we adopt the ID from the
+        // configuration object and assume that the types will be the same
+        target.id = configuration.id;
+
+        const context = { synchronize };
+
+        synchronize(target, configuration, context);
+        // TODO: layers don't render on initialization in some cases, needs investigation
+        mapElement.configuration.updateSize();
+      },
+    };
+
+    mapElement.configuration.on("moveend", (_event) => {
```

Comment:
It's possible for `mapElement.configuration` to be `undefined` at this point. For example, if you don't have a Map on the page beforehand and add one inside a Dialog (gets first added as a virtual child).

The error has to do with init timing in the `<vaadin-map>` Web Component: `configuration` property should be available right after the constructor has run, (`ready()` is too late since it will invoke only once the component class has finalized)

Reply:
Great find, opened a bug ticket for now: https://github.com/vaadin/flow-components/issues/2595